### DISCUSSION TITLE
Import and export as text

### DISCUSSION THREAD

**Discussion by anjanai** [Studio-kalavati/Swaranotebook-bandish-editor#113] [Ideas / Feature Requests]
Would be nice to be able to import and export from and to text format.

**Comment by shark8me** [comment]
Unfortunately it is not straightforward to represent a Bandish with text,
because the vertical alignment of Svaras, lyrics and taal sam/khaali isn't
possible without special fonts or Latex.
The app uses images for the Svaras (which makes it easy to support multiple
languages, but hard to use in a utf-8 or unicode text editor).

The notation is saved as a json file and can be exported as PDF, if
that helps.

Importing may be possible, are you aware of text formatted example of a
Bandish?

**Comment by anjanai** [comment]
Thanks, Kiran.

Yes, I understand the concerns.

It's possible to notate a bandish in text form and then import it. Of
course, you will need to define a certain standard and expect people to
stick to it. I am attaching a text file. I have used it to notate some
snippets of compositions that I could then play online.

See https://anjanai.github.io/music-notation/compositions.html - click on
the word "example" on that page, and then press the Play button on the page
that pops up. Also, at the bottom of that page, you will find a link to
download a wav file. Try that as well.

This example file contains a single line of many different bandishes in
different raags - all from the book
https://archive.org/details/hindustani-sangeet-paddhati-kramik-pustak-malika-part-1/hindustani-sangeet-paddhati-kramik-pustak-malika-part-1/
.

Having said that it is possible, it might not be something that is going to
be used by too many people, so might not be worth the effort.

Anyway, thanks for considering it.

Regards,
-Anjana


On Fri, Jan 26, 2024 at 10:34 AM Kiran Karkera ***@***.***>
wrote:

> Unfortunately it is not straightforward to represent a Bandish with text,
> because the vertical alignment of Svaras, lyrics and taal sam/khaali isn't
> possible without special fonts or Latex.
> The app uses images for the Svaras (which makes it easy to support
> multiple
> languages, but hard to use in a utf-8 or unicode text editor).
>
> The notation is saved as a json file and can be exported as PDF, if
> that helps.
>
> Importing may be possible, are you aware of text formatted example of a
> Bandish?
>
>
> On Thu, Jan 25, 2024, 5:30 AM anjanai ***@***.***> wrote:
>
> > Would be nice to be able to import and export from and to text format.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Studio-kalavati/bandish-editor/discussions/113>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAAVX5EJOELORTIHGMEHUS3YQGODRAVCNFSM6AAAAABCJTQPH2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGEZDSMJTGI>
>
> > .
> > You are receiving this because you are subscribed to this thread.Message
> > ID: ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Studio-kalavati/bandish-editor/discussions/113#discussioncomment-8260553>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJPP4GSEM2TWXE2WPKJTZFLYQPZL7AVCNFSM6AAAAABCJTQPH2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENRQGU2TG>
> .
> You are receiving this because you authored the thread.Message ID:
> <Studio-kalavati/bandish-editor/repo-discussions/113/comments/8260553@
> github.com>
>

L: 1/4
M: 4/4
Q:1/4=150 
K: C^f
P: Yaman : सदा शिव भज मना निस दिन 
srg: | n d - p | m p g m || p - - - | p m g r |
w: स दा शि व भ ज म ना नि स दि न

K: C
P: Bilawal :  तू ही आधार सकल त्रिभुवन को
srg: | g p n n || S - S S | S R S n | d p mg rs |
w:  तू  ऽ ही आ | धा  र स क ल त्रि भु व न को  ऽ  ऽ ऽ

K: C_b
P: Khamaj :  पनघट मुरलिया बाजे सखी
srg: S | n n dp m | p m g g || m - p m | p - - - | 
w:  प न घ ट ऽ  मु र लि ऽ या     बा जे स खी

K: C_d_a
P: Bhairav :  जागो मोहन प्यारे 
srg: | g m d - | p - d m || dd pm p - | m - g - |
w:  जा ऽ गो | मो ह न | प्या ऽ ऽ ऽ ऽ रे ऽ 

K: C_d^f_a
P: Poorvi : ए री मैका सब सुख दीनी
srg: | n, r g p- | p p - p || p d m pm | g =m g - | 
w: | ए ऽ री ऽ | ~ मै का | स ब सु ख ऽ दी ऽ नी

K: C_d^f
P: Marwa : लई रे श्याम हमरी डुरियाँ
srg: | d m - g- | g r - s || n, r g - | m m d - | 
w: ल ई रे        ऽ श्या म | ह म री |   डु रि याँ

K: C_e_b
P: Kafi: प्रभु तेरी दया है अपार
srg: | s n, | s r r g | - m - m || p - p m | p d n 
w: प्र भु ते री द या       ऽ है अ    पा र तू अ ग म 

K: C_e_a_b
P: Asavari: हरि बिन तेरा कौन सहायी
srg: | d m pd mp | g - r s | r m p S || d - p - |
w: ह रि बि ऽ न ऽ  ते रा ऽ | कौ ऽ न स | हा यी

K: C_d_e_a_b
P: Bhairavi: बाट चलत मोरी चुनरी रंग डारी
srg: | p - p d | m p g m || p n d p | m g r s |
w: बा ट च | ल त मो री | चु न री रं | ग डा ऽ री |

K: C_d_e^f_a
P: Todi: नेक चाल चलिए चतुर
srg: n, s g m | p m p d || m - g r | g r s s |
w: ने ऽ क चा |  ऽ ल च लि | ए  ऽ ऽ |  ऽ च तु र

**Reply by shark8me** [reply]
If I understand correctly, this is written using the `abc` notation. 
Since the `abc` notations uses absolute pitch (like other western music notations), the Sa/Shadja would have to be indicated and the notes in the rest of the composition 'translated' <using this word informally> from absolute pitch to relative pitch. 

While it is possible to build an abc to Bandish notation converter, I'm not sure it'll be widely used unless there's a large corpus of Bandishes already written in `abc` notation format. 

<digression>
A MIDI to Bandish translator may be a better idea because it is easier for someone to play a Bandish on a keyboard, save the notes as a MIDI file and later view the notes. 
</digression>
